flux_led/protocol.py

```diff
@@ -1378,6 +1403,89 @@ def construct_preset_pattern(
+        """The name of the protocol."""
+        return PROTOCOL_LEDENET_25BYTE
+
+    def construct_levels_change(
+        self,
+        persist: int,
+        red: int | None,
+        green: int | None,
+        blue: int | None,
+        warm_white: int | None,
+        cool_white: int | None,
+        write_mode: LevelWriteMode | int,
+    ) -> list[bytearray]:
+        """The bytes to send for a level change request."""
+        # sample message for 25-byte LEDENET protocol (w/ checksum at end)
+        #  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23
+        # b0 b1 b2 b3 00 01 02 79 00 0e e0 01 00 a1 78 64 64 00 00 00 00 14 00 00
+        #                    |  |              |  |  |  |  |  |  |              |
+        #                    |  |              |  |  |  |  |  |  |              footer
+        #                    |  |              |  |  |  |  |  |  white brightness
+        #                    |  |              |  |  |  |  |  white temperature (00 warm - 64 cool)
+        #                    |  |              |  |  |  |  value
+        #                    |  |              |  |  |  saturation
+        #                    |  |              |  |  hue / 2
+        #                    |  |              |  write mode (a1 color, b1 white)
+        #                    |  |              unknown filler
+        #                    |  increment counter
+        #                    header
+
+        if red is not None and green is not None and blue is not None:
+            h, s, v = colorsys.rgb_to_hsv(red / 255, green / 255, blue / 255)
+            h = int((h * 360) / 2)  # Hue needs to be halved
+            s = int(s * 100)
+            v = int(v * 100)
+        else:
+            h = s = v = 0x00
+
+        if warm_white is not None and cool_white is not None:
+            # warm white comes through as 0, even when temp is set to 6500
+            white_brightness = round(((warm_white + cool_white) / (2 * 255)) * 64)
+            white_temp = round((cool_white / (warm_white + cool_white)) * 64)
+        elif cool_white is not None and warm_white is None:
+            white_temp = 0x64
+            white_brightness = int((cool_white / 255) * 100)
+        else:
+            white_temp = 0x00
+            white_brightness = 0x00
+
+        return [
+            self.construct_wrapped_message(
+                bytearray(
+                    [
+                        0xE0,
+                        0x01,
+                        0x00,
+                        write_mode.value
+                        if isinstance(write_mode, LevelWriteMode)
+                        else write_mode,
+                        h,
+                        s,
+                        v,
+                        white_temp,
+                        white_brightness,
+                        0x00,
+                        0x00,
+                        0x14,
+                        0x00,
+                        0x00,
```

Comment:
I was a bit surprised that the checksum is always. 0x00 on the inner message.

I wonder if we can remove `inner_pre_constructed=True` and drop the last byte here

Reply:
tested, doesn't work